lib/oauth.js

```diff
@@ -167,9 +167,17 @@ exports.v2 = function (settings) {
             }
 
             query.client_id = settings.clientId;
-            query.response_type = 'code';
+            query.response_type = settings.provider.response_type;
             query.redirect_uri = internals.location(request, protocol, settings.location);
             query.state = nonce;
+            if (settings.provider.nonce) {
+                query.nonce = nonce; // required by microsoftonline
```

Comment:
For me it looks like a rudimental property left by MS. It does not matter, what nonce would be sent, they return `state` property anyway, on which we make a compare later. But both `nonce` and `state` are required anyway. Personally for me it looks like a bug.